### PR TITLE
only replace note ID when note ID starts with `@` character

### PR DESCRIPTION
**Pull request by SamSamskies**
resolves https://github.com/v0l/snort/issues/440

Screenshot 2023-03-14 at 3 39 16 PM

```json
{
  "id": "d098bbfc7fb36cc5204012ad5db7b624e77f934f98cfd39301409c7838e0cca3",
  "pubkey": "03e20803e8917f3ba4e029e6fcd2a3c1b97cdd7c691119c1bdf4bb7543e4e1c1",
  "created_at": 1678826024,
  "kind": 1,
  "tags": [
    [
      "e",
      "d5dc392fc5b381c526182ffdfda0ddfc4eca0b67eedbb4179bfc9f1d9bb3c7c0",
      "",
      "mention"
    ]
  ],
  "content": "Note IDs should be replaced with a tag: \n#[0]\nOther text such as link that contain note IDs should not be replaced with a tag:\n\nhttps://snort.social/e/note16hwrjt79kwqu2fsc9l7lmgxal38v5zm8amdmg9umlj03mxanclqqn7k634\n\n",
  "sig": "4e5166ff2a0a78679b01b1e71c35522cbfc481bfa76336f0ca9b938ced463a918e29966ea3d39745bc0a550ace4f1dc7ea0c46c20cace5e83f9843590707b8a4",
  "relays": [
    "wss://nostr-relay.alekberg.net"
  ]
}
```

### CODE REVIEW

**Comment by SamSamskies** [comment]
actually new lines were removed when they shouldn't have been. turning back into draft.

File: packages/app/src/Feed/EventPublisher.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export default function useEventPublisher() {
     };
     const content = msg
       .replace(/@npub[a-z0-9]+/g, replaceNpub)
-      .replace(/note1[acdefghjklmnpqrstuvwxyz023456789]{58}/g, replaceNoteId)
+      .replace(/(\s)(note1[acdefghjklmnpqrstuvwxyz023456789]{58})(\s)/g, replaceNoteId)
```
Comment (v0l) [inline packages/app/src/Feed/EventPublisher.ts:79]: What if i wanted to start the note with a ref?

Comment (SamSamskies) [reply packages/app/src/Feed/EventPublisher.ts:79]: do you mean like this `@note1ju6usp2u7d5v6uxxka2d37p5p400djuu555yp46h6qdrnf85k5kqg49a8c`?

Comment (v0l) [reply packages/app/src/Feed/EventPublisher.ts:79]: Yep

File: packages/app/src/Feed/EventPublisher.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ export default function useEventPublisher() {
     };
     const content = msg
       .replace(/@npub[a-z0-9]+/g, replaceNpub)
-      .replace(/note1[acdefghjklmnpqrstuvwxyz023456789]{58}/g, replaceNoteId)
+      .replace(/(\s)@{0,1}(note1[acdefghjklmnpqrstuvwxyz023456789]{58})(\s)/g, replaceNoteId)
```
[the remark this answers is not in the document]
Comment (SamSamskies) [reply packages/app/src/Feed/EventPublisher.ts:79]: ah okay. that might mess some people up though cause it's always worked without a leading `@` on Snort. i guess people will learn though.

[the remark this answers is not in the document]
Comment (v0l) [reply packages/app/src/Feed/EventPublisher.ts]: Note creator really needs some help tips or something, for auto-complete etc

**Review by v0l — APPROVED** [review]

**Pull request merged**